tests/test_parser.py

```diff
@@ -0,0 +1,60 @@
+import os
+
+import pandas as pd
+import pytest
+
+
+def test_parse_product(parser, mock_data):
+    """Test parse product method"""
+    product_filepath = os.path.join(mock_data, "mock_prefix_products_20210221.csv")
+    product_df = pd.read_csv(
+        product_filepath, sep=";", dtype={"SKU": str, "BARCODES": str}
+    )
+    parsed_result_df = parser.parse_product(product_df)
+    actual_product_filepath = os.path.join(mock_data, "processed_products.csv")
+    actual_product_df = pd.read_csv(
+        actual_product_filepath, sep=";", dtype={"sku": str}
+    )
+    pd.testing.assert_frame_equal(parsed_result_df, actual_product_df, check_like=True)
+
+
+def test_validate_products(parser, mock_data):
+    """Test that validate products method raises exceptions when needed"""
+    product_filepath = os.path.join(mock_data, "bad_products.csv")
+    product_df = pd.read_csv(product_filepath, sep=";", dtype={"sku": str})
+    for index, row in product_df.iterrows():
+        with pytest.raises(ValueError):
+            parser._validate_products(
+                pd.DataFrame(data=dict(zip(row.index, row.values)), index=[0])
+            )
+
```

Comment:
I don't follow this. why not do it like this:
```suggestion
    with pytest.raises(ValueError):
        parser._validate_products(product_df)
```

Reply:
If I use what you day then it will only raise the error for the first invalid row.
The way I did it, we get the assertion for each bad row of data in the csv file.

Reply:
I got it now, thanks.